packages/app/src/client/views/login/index.tsx

```diff
@@ -175,17 +259,44 @@ const Login = (props: { state?: State }) => {
   );
 };
 
-const LoginPage = () => {
+function LoginPageParent() {
+  // Uninitialized state will cause Child to error out
+  const [loginConfig, setLoginConfig] = useState<LoginConfigInterface | undefined>();
+  useEffect(() => {
+    (async () => {
+      // Without this early exit criterion, `fetchLoginConfig()` is going to
+      // be called with every render cycle.
+      if (loginConfig !== undefined) {
+        return
+      }
+
+      const lcfg = await fetchLoginConfig();
+      setLoginConfig(lcfg);
+    })();
+  });
+
+  // Do not render child until loginconfig is populated. Kudos to
```

Comment:
_This_ concept is the sugar here. Thanks to https://daveceddia.com/react-before-render/.

Reply:
This is a pattern you'll see used in React (and already in our UI) many times 👍